packages/cli-repl/src/cli-repl.ts

```diff
@@ -243,17 +244,23 @@ class CliRepl {
 
     try {
       fd = fs.openSync(configPath, 'wx');
-      this.userId = new ObjectId(Date.now());
+      this.userId = new ObjectId(Date.now()).toString();
       this.enableTelemetry = true;
       this.disableGreetingMessage = false;
       this.bus.emit('mongosh:new-user', this.userId, this.enableTelemetry);
       this.writeConfigFileSync(configPath);
     } catch (err) {
       if (err.code === 'EEXIST') {
-        const config = JSON.parse(fs.readFileSync(configPath, 'utf8'));
-        this.userId = config.userId;
+        const config = jsonParse(fs.readFileSync(configPath, 'utf8'));
```

Comment:
I think a comment explaining why we are not using `JSON.parse` here is necessary. I understand you want a tolerant version of JSON.parse that does not throw, but is not clear from this code.